ui/src/pages/backoffice/DocumentRequest/DocumentRequestDetails/components/DocumentRequestSteps/components/DocumentProvidersStep.js

```diff
@@ -0,0 +1,110 @@
+import { ESSelector } from '@components/ESSelector';
+import {
+  serializeAcqOrder,
+  serializeBorrowingRequest,
+} from '@components/ESSelector/serializer';
+import { goTo } from '@history';
+
+export class DocumentProvidersStep extends Component {
+  onCreateAcq() {
+    return goTo(AcquisitionRoutes.orderCreate);
+  }
+
+  onCreateIll() {
+    return goTo(ILLRoutes.borrowingRequestCreate);
+  }
+
+  onAcqSelectResult = async data => {
+    // TODO: Fire a request to update document request with provider info
+    // const resp = await documentRequestApi.accept(pid, {
+    //   provider: {type: "ILL or ACQ", value: data.pid},
+    // });
+
+    // NOTE: Redirect back here to continue with the next step
+    // if (resp.status === 202) {
+    //   this.props.fetchDocumentRequestDetails(pid);
+    // }
+    console.log('Selected ACQ data', data);
+  };
+
+  onIllSelectResult = async data => {
+    // NOTE: do the same things as onAcqSelectResult
+    console.log('Selected ILL data', data);
+  };
+
+  renderAcqSegment() {
+    return (
+      <Segment raised>
+        <Label color="grey" ribbon>
+          Acquisition
+        </Label>
+        <span>Search and select an existing Acquisition order</span>
+        <Grid columns={2} padded>
+          <Grid.Column>
+            <ESSelector
+              onSelectResult={this.onAcqSelectResult}
+              query={acqOrderApi.list}
+              serializer={serializeAcqOrder}
+            />
+          </Grid.Column>
+          <Grid.Column textAlign="center" verticalAlign="middle">
+            <Button
+              positive
+              style={{ width: '300px' }}
+              name="create-acq"
+              onClick={this.onCreateAcq}
+              icon={'plus'}
+              content={'Create new Acquisition Order'}
+            />
+          </Grid.Column>
+        </Grid>
+        <Divider vertical>Or</Divider>
+      </Segment>
+    );
+  }
+
+  renderIllSegment() {
+    return (
+      <Segment raised>
+        <Label color="grey" ribbon>
+          Interlibrary
+        </Label>
+        <span>Search and select an existing Inter Library loan</span>
+        <Grid columns={2} padded>
+          <Grid.Column>
+            <ESSelector
+              onSelectResult={this.onIllSelectResult}
+              query={borrowingRequestApi.list}
+              serializer={serializeBorrowingRequest}
+            />
+          </Grid.Column>
+          <Grid.Column textAlign="center" verticalAlign="middle">
+            <Button
+              positive
+              style={{ width: '300px' }}
+              name="create-ill"
+              onClick={this.onCreateIll}
+              icon={'plus'}
+              content={'Create new Interlibrary Loan'}
+            />
+          </Grid.Column>
+        </Grid>
+        <Divider vertical>Or</Divider>
+      </Segment>
+    );
+  }
+
+  render() {
+    return (
+      <>
+        {this.renderAcqSegment()}
```

Comment:
do you think wrapping them in methods is better for readability? for me more natural is to follow the lines instead of jumping to the method. @zzacharo @equadon what do you think? (if there is not a lot of code, when there is more it should probably be split for smaller components